### PR TITLE
[CLOUD-5932]: android local storage migration

### PR DESCRIPTION
**Pull request by jahead**

### CODE REVIEW

File: src/android/StorageMigrator.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    private void commitToNativeStorage(Map<String, String> keyValues, SharedPreferences.Editor editor) {
+        Log.d(TAG, "commitToNativeStorage: Starting to write to shared pref");
+
+        for (Map.Entry<String, String> entry : keyValues.entrySet()) {
+            String key   = entry.getKey();
+            String value = entry.getValue();
+
+            if(!CONVERSION_MAP.containsKey(key)) {
```
Comment (franco-octavianus) [inline src/android/StorageMigrator.java]: does this match `rapid-rma-cognito-device-key.<username>`?

File: src/android/StorageMigrator.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    public boolean hasMigrated() {
+        return sharedPref.contains("rapid-username");
+    }
+
+    public boolean migrate() {
+        Log.d(TAG, "migrate: Checking is migration has already run");
+        if(this.hasMigrated()) {
+            Log.d(TAG, "migrate: hasMigrated() return false");
```
Comment (franco-octavianus) [inline src/android/StorageMigrator.java]:
```suggestion
            Log.d(TAG, "migrate: hasMigrated() return true");
```

File: src/android/StorageMigrator.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            return true;
+        }
+        Log.d(TAG, "migrate: hasMigrated() return true, starting migration");
```
Comment (franco-octavianus) [inline src/android/StorageMigrator.java]:
```suggestion
        Log.d(TAG, "migrate: hasMigrated() return false, starting migration");
```

**Comment by franco-octavianus** [comment]
@jahead are we still going to clean up the original local storage by removing the migrated key values?

**Comment by jahead** [comment]
> @jahead are we still going to clean up the original local storage by removing the migrated key values?

I want to but I was going to do that in a seperate pr. As that's a different unit of work. 

But i wasn't sure what you were going to do on the iOS side

**Comment by franco-octavianus** [comment]
> > @jahead are we still going to clean up the original local storage by removing the migrated key values?
> 
> I want to but I was going to do that in a seperate pr. As that's a different unit of work.
> 
> But i wasn't sure what you were going to do on the iOS side

iOS side has done the clean up:
https://github.com/ischemaview/cordova-plugin-nativestorage/pull/2/files#diff-9e88ac2f85f6b093c0241f5212a38b948a3a4ec6dc436993bcb800c77084c048R169

**Comment by jahead** [comment]
Awesome, I'll do the pr this morning cann you review this so I can base it off this one?

**Review by franco-octavianus — APPROVED** [review]

**Review by franco-octavianus — APPROVED** [review]

**Pull request merged**